### PR TITLE
Fix CI jobs

### PR DESCRIPTION
**Pull request by evgeniy-scherbina**
Changes:
- update go version to 1.21 in Github Actions jobs
- fix v47 bug:
  - using `genutil.NewAppModuleBasic(genutiltypes.DefaultMessageValidator)` instead of `genutil.AppModuleBasic{}`
 - set legacy router for gov module:
   - `govKeeper.SetLegacyRouter(govRouter)`
 - Set chainID:
   - `baseapp.SetChainID(chainID)`
 - Downgrade golang in go.mod:
   - `go1.21.6` -> `go1.21.0` to avoid nix-related errors
 - Regenerate `gomod2nix.toml`
   - it fails to do it in regular way:
     - `nix-shell -I nixpkgs=./nix ./tests/integration_tests/shell.nix`
     - `gomod2nix generate` (inside nix-shell)
     - error: `panic: error generating pkgs: go.mod:5: unknown directive: toolchain`;
     - gomod2nix doesn't work with go1.21 out of the box - need some additional investigations
   - so I used alternative approach from here: https://github.com/nix-community/gomod2nix/issues/134#issuecomment-1898694919 
   -  but it worth spending time to fix error above
 - looks like `testenv.nix` ignores version of `poetry2nix` which specified in `sources.json`, so I had to explicitly pass it to `testenv.nix`:
   - `(_: pkgs: { test-env = pkgs.callPackage ./testenv.nix { poetry2nix = import sources.poetry2nix {}; }; })`
 - change `buildGo120Module -> buildGo121Module`
 - Updated `sources.json`:
   - Updated version of nixpkgs to https://github.com/NixOS/nixpkgs/releases/tag/23.11 (which include go1.21)
 - Change ethermint version in `tests/integration_tests/configs/upgrade-test-package.nix` which affects `test_cosmovisor_upgrade`
 - Fix `tests/integration_tests/test_grpc_only.py`, partially copy-pasted from cronos's ethermint.
 - Fix `tests/integration_tests/test_upgrade.py`
 
 `NOTE`:
Steps to reproduce regenerating of `gomod2nix.toml`:
- create `flake.nix` in root of ethermint, source code: https://github.com/nix-community/gomod2nix/issues/134#issuecomment-1898694919
- run: `nix develop --extra-experimental-features nix-command --extra-experimental-features flakes`
- run `gomod2nix` inside develop shell
- it worth spending time to look into `nix flake`

`UPD`:
- Issue with regeneration of `gomod2nix.toml` was resolved

### CODE REVIEW

File: tests/integration_tests/test_grpc_only.py [hunk 1]
```diff
@@ -100,29 +126,29 @@ def test_grpc_mode(custom_ethermint):
             wait_for_port(grpc_port)
             wait_for_port(api_port)
 
-            # in grpc-only mode, grpc query don't work if we don't pass chain_id
-            rsp = grpc_eth_call(api_port, msg)
-            assert rsp["code"] != 0, str(rsp)
-            assert "invalid chain ID" in rsp["message"]
+            def expect_cb(rsp):
```
Comment (drklee3) [inline tests/integration_tests/test_grpc_only.py:129]: nice fixes, was a finicky test

**Review by drklee3 — APPROVED** [review]

File: .github/workflows/lint.yml
[the remark this answers is not in the document]
Comment (evgeniy-scherbina) [reply .github/workflows/lint.yml:1]: FIxed based on this example: https://github.com/crypto-org-chain/ethermint

File: tests/integration_tests/test_grpc_only.py (hunk 1 above)
Comment (evgeniy-scherbina) [reply tests/integration_tests/test_grpc_only.py:129]: I was inspired by https://github.com/crypto-org-chain/ethermint

**Review by nddeluca — APPROVED** [review]
Great work, changes look good

**Pull request merged**